code/game/objects/items/tanks/tanks.dm

```diff
@@ -44,37 +44,37 @@
 	var/list/explosion_info
 	/// List containing reactions happening inside our tank.
 	var/list/reaction_info
+	/// Mob that is currently breathing from the tank.
+	var/mob/living/carbon/breathing_mob = null
 
-/obj/item/tank/ui_action_click(mob/user)
-	toggle_internals(user)
+/// Closes the tank if dropped while open.
+/obj/item/tank/dropped(mob/living/user, silent)
+	. = ..()
+	// Close open air tank if it got dropped by it's current user.
+	if (breathing_mob && (loc != breathing_mob))
+		breathing_mob.cutoff_internals()
 
-/obj/item/tank/proc/toggle_internals(mob/user)
-	var/mob/living/carbon/human/H = user
-	if(!istype(H))
-		return
+/// Closes the tank if given to another mob while open.
+/obj/item/tank/equipped(mob/living/user, slot, initial)
+	. = ..()
+	// Close open air tank if it was equipped by a mob other than the current user.
+	if (breathing_mob && (user != breathing_mob))
+		breathing_mob.cutoff_internals()
```

Comment:
it would be nice to handle this with signals, mostly to avoid the hardref

Reply:
Fixed by commit 4ecc5a90b43978f60ecb50e92a76e0fa900015ad